eventrouter/internal/eventrouter_freertos.c

```diff
@@ -457,22 +454,7 @@ void ErReturnToSender(ErEvent_t *a_event)
 
     const int previous_reference_count =
         atomic_fetch_sub(&a_event->m_reference_count, 1);
-    const int reference_count     = previous_reference_count - 1;
-    const size_t sending_task_idx = a_event->m_sending_module->m_task_idx;
-    const ErTask_t *sending_task =
-        &s_context.m_options->m_tasks[sending_task_idx];
-
-    // Clients should not call `ErReturnToSender()` on an event in the same
-    // handler call that they receive it; it messes up the reference count and
-    // they can do the same thing by returning `ER_EVENT_HANDLER_RET__HANDLED`.
-    //
-    // This constraint prevents subscribers from KEEPing events which are sent
-    // multiple times using `ErSendEx()`. Without this check a subscriber could
-    // receive an event which it had previously KEPT, return the previously KEPT
-    // copy, and then KEEP the new copy. I can't see how this flow could be
-    // useful and think that preventing incorrect calls to `ErReturnToSender()`
-    // is more useful.
```

Comment:
I should have seen how this could be useful, given that I already rely on it to make re-sendable/cancelable timers work.